src/mmap_storage.cpp

```diff
@@ -161,15 +162,22 @@ error_code translate_error(std::error_code const& err, bool const write)
 
 			download_priority_t const old_prio = m_file_priority[i];
 			download_priority_t new_prio = prio[i];
+
+			m_file_priority[i] = new_prio;
```

Comment:
Again, I'm not a C++ guy, but this looks sketchy to me. Also considering that `new_prio` is not const. Why we need intermediate variable?

Reply:
it's not necessary. but I like giving it an intuitive name. it probably should be `const`, but it's a copy either way.